doc/development/guide/deprecations_removals.rst

```diff
@@ -0,0 +1,85 @@
+Deprecations and Removals
+=========================
+
+PennyLane is a rapidly-evolving project, and the details of functions, Operators & more are
+subject to change. These changes ensure that PennyLane is the best that it can be, but they will
+sometimes break existing conventions. When these breaking changes are necessary, we should make
+sure to give our users time to update their workflows to adhere to any new implementation before
+completely removing the old one.
+
+Deprecating a feature
+---------------------
+
+The means of informing users of a breaking change is called a deprecation cycle. Before removing a
+feature, for one or more releases of PennyLane, the deprecated feature should advise users of the
+upcoming change while preserving its functionality. Here are the steps needed to properly deprecate
+a feature:
+
+1. Identify the new or preferred way of achieving the same functionality once the deprecated
+   feature is removed. The exception to this is when a feature is being removed for lack of use.
+
+2. At the beginning of the relevant code (eg. the ``__init__()`` method of a class that's being
+   removed), add the following lines of code, filling in the relevant details:
+
+   .. code-block:: python
+
+       warnings.warn(
+           "<name-of-feature> is deprecated, and will be removed in an upcoming release. "
+           "Instead, please use <name-of-preferred-way-to-achieve-functionality>.",
+           qml.PennyLaneDeprecationWarning,
+       )
+
+   If the feature is being relocated, consider rephrashing the warning to discuss relocation as
+   opposed to deprecation.
+
+3. If the feature has public-facing docs, include a similar warning message in a visible part of
+   its docstring using a ``.. warning::`` Sphinx directive.
+
+4. Replace all uses of the deprecated code within PennyLane's own source code. This is to ensure
+   that the warning isn't unexpectedly raised for users that did not personally call the deprecated
+   piece of code.
+
+5. Remove any tests that specifically test the deprecated code. For tests that depend it but are
+   not written to test it specifically, update them to use the new/preferred code.
```

Comment:
Should we be removing the tests at the deprecation stage, or the removal stage?

Reply:
it's probably wiser to wait until it's removed, will update